homeassistant/components/bluetooth/scanner.py

```diff
@@ -218,13 +230,12 @@ def _async_detection_callback(
 
     async def async_start(self) -> None:
         """Start bluetooth scanner."""
-        self.scanner.register_detection_callback(self._async_detection_callback)
-
         async with self._start_stop_lock:
             await self._async_start()
 
     async def _async_start(self) -> None:
         """Start bluetooth scanner under the lock."""
+        assert self.scanner is not None
```

Comment:
You could declare the variable as a class variable without initialization, then just assign in setup. Since setup is now a required call. I dont really see why though. Its not async, so you could do it in the initializer.